### PR TITLE
Added docs for reverse proxy to README

### PR DESCRIPTION
**Pull request by tk-nguyen**
As mentioned in https://github.com/tweedegolf/mailcrab/issues/55#issuecomment-1538219800, I've included example configurations for nginx and apache2.

### CODE REVIEW

**Comment by stappersg** [comment]
From my _sent items_
```text
Date: Sun, 14 May 2023 18:09:12 +0200
From: Geert Stappers <stappers@stappers.it>
To: tweedegolf/mailcrab <noreply@github.com>
Subject: Re: [tweedegolf/mailcrab] Added docs for reverse proxy to README (PR #60)

On Sun, May 14, 2023 at 06:58:01AM -0700, Nguyen Thai wrote:
> @tk-nguyen pushed 1 commit.
> 10d4b605e7e9f41f6bdf53268e14bc3834ee11fe  Added docs for reverse proxy to README
> Message ID: <tweedegolf/mailcrab/pull/60/push/13629357431@github.com>

- Thanks for the merge request
- The Apache (>= 2.4.47) configuration looks different from what I reported in #55
- Text 'also enable `mod_proxy_wstunnel`.' belongs to 2.4.46 section
- Maybe replacing the Apache 2.4.46 section with "older version need
  `mod_proxy_wstunnel`"
- Move the '### Reverse proxy' paragraph below '### Path prefix'
  paragraph, or even to a `Reverse_proxy.md` file.


Regards
Geert Stappers
Fully aware that he could make a merge request himself.
Fully aware that he should make a merge request himself.
--
Silence is hard to parse
```

**Comment by tk-nguyen** [comment]
> - The Apache (>= 2.4.47) configuration looks different from what I reported in #55

For the Apache (>= 2.4.47) configuration, I'm following the [docs of `ProxyPass`](https://httpd.apache.org/docs/2.4/mod/mod_proxy.html#protoupgrade), like [in the examples](https://httpd.apache.org/docs/2.4/mod/mod_proxy.html#wsupgrade).

> - Text 'also enable `mod_proxy_wstunnel`.' belongs to 2.4.46 section
> - Maybe replacing the Apache 2.4.46 section with "older version need
  `mod_proxy_wstunnel`"

Older versions have a different configuration, so I'm thinking of merging both of these.

> - Move the '### Reverse proxy' paragraph below '### Path prefix'
  paragraph, or even to a `Reverse_proxy.md` file.

Good point, I can also include more explanations, then link to this from README.

**Comment by marlonbaeten** [comment]
Nice work @tk-nguyen!
I'll wait for any changes you mentioned have before merging.

**Comment by tk-nguyen** [comment]
I've splited the guide to a seperate file (`Reverse_proxy.md`) and expanded the docs.

**Comment by marlonbaeten** [comment]
Awesome - thanks!
